src/Database/Expression/CaseExpression.php

```diff
@@ -130,6 +130,10 @@ protected function _addExpressions($conditions, $values, $types)
                 $value = $keyValues[$k];
                 array_push($this->_values, $value);
                 continue;
+            } elseif ($value === 'identifier') {
+                $value = new IdentifierExpression($keyValues[$k]);
+                array_push($this->_values, $value);
+                continue;
```

Comment:
Can we add this case here too? https://github.com/cakephp/cakephp/blob/master/src/Database/Expression/FunctionExpression.php#L110